client/src/components/user/dashboard.js

```diff
@@ -1,39 +1,265 @@
+    } else {
+      display = <DashboardCalendar subscriptions={subscriptions} />;
+    }
+
+    return display;
   }
 
   render() {
-    const { successMessage } = this.state;
+    const { pfp } = this.props;
+    const { subscriptions, addingSubscription, editingSubscription } = this.state;
+
+    const subscriptionForm = addingSubscription
+      ? <div className="p-3 m-2 d-flex flex-wrap borderSubscriptionForm">
+          <div className="col d-flex justify-content-between align-items-center">
+            <div />
+            <h2 className="text-start">Create Subscription</h2>
+            <button className="btn btn-link my-2 d-md-none" type="button" data-bs-dismiss="offcanvas" aria-label="Close" onClick={() => this.setState({ addingSubscription: !addingSubscription })}>
+              <FontAwesomeIcon icon={faTimes} />
+            </button>
+            <button className="btn btn-link my-2 d-none d-md-block" type="button" onClick={() => this.setState({ addingSubscription: !addingSubscription })}>
+              <FontAwesomeIcon icon={faTimes} />
+            </button>
+          </div>
+          <CreateSubscription 
+          addSubscription={this.handleUpdate}
+          toggleLoadingState={this.toggleLoadingState}
+          showSubscriptionList={this.showSubscriptionList}
+          currentSubscriptions={subscriptions} />
+        </div>
+      : <div className="p-3 m-2 d-flex flex-wrap borderSubscriptionForm">
+          <div className="col d-flex justify-content-between align-items-center">
+            <div />
+            <h2 className="text-start">Update Subscription</h2>
+            <button className="btn btn-link my-2 d-md-none" type="button" data-bs-dismiss="offcanvas" aria-label="Close" onClick={() => this.setState({ editingSubscription: !editingSubscription, activeSubscription: false })} >
+              <FontAwesomeIcon icon={faTimes} />
+            </button>
+            <button className="btn btn-link my-2 d-none d-md-block" type="button" onClick={() => this.setState({ editingSubscription: !editingSubscription, activeSubscription: false })}>
+              <FontAwesomeIcon icon={faTimes} />
+            </button>
+          </div>
+          <UpdateSubscription
+          updateSubscription={this.handleUpdate}
+          showSubscriptionList={this.showSubscriptionList}
+          toggleLoadingState={this.toggleLoadingState}
+          prevSubscription={editingSubscription}
+          />
+        </div>;
     return (
-      <div>
-        <header>
-          <h1>Hello this is the users dashboard page.</h1>
-          <section className="subscription-list">
-            <SubscriptionsList />
-          </section>
-          {successMessage}
-          <a href={href}>
-            Sign Out!
-            <FontAwesomeIcon icon={faSignOutAlt} />
+      <div className="h-100 d-flex flex-column">
+        <header className="navbar p-2 d-flex justify-content-between align-items-center text-dark border-bottom shadow-sm">
```

Comment:
i think this could use a bit more horizontal padding on desktop

```suggestion
        <header className="navbar py-2 px-4 d-flex justify-content-between align-items-center text-dark border-bottom shadow-sm">
```

Reply:
And on mobile, we can maybe shrink the header size so that the icon, date, and pfp all fit in one line since right now it wraps:
<img width="448" alt="Screen Shot 2022-03-23 at 1 11 50 AM" src="https://user-images.githubusercontent.com/25065329/159628398-e576f04e-d251-41b6-8a5b-de2163eb1fdc.png">
 
But if you did say `h3` as the class for the header it would look like this:
<img width="448" alt="Screen Shot 2022-03-23 at 1 13 21 AM" src="https://user-images.githubusercontent.com/25065329/159628561-cae313f5-1d1d-4a33-8834-19fa38917bc5.png">